stablehlo/reference/Ops.cpp

```diff
@@ -1547,6 +1560,14 @@ Tensor evalRealOp(const Tensor &operand, ShapedType resultType) {
   return result;
 }
 
+SmallVector<InterpreterValue> evalRecvOp(Token token, ChannelId channelId,
+                                         Process *process) {
+  SmallVector<InterpreterValue> results;
+  for (const auto &tensor : process->recv(channelId)) results.push_back(tensor);
+  results.push_back(token);
```

Comment:
What happens if send/recv have different types, i.e. send puts 3 tensors, but receive expects 4? Does this type checking happen elsewhere in the interpreter?

Reply:
This should be caught when trying to add incompatible types to the Scope object through an assertion (e.g. Scope expects `Value` with 4 but the Tensor value only has 3)l